MRTK3 Tutorials/Assets/MRTK.Tutorials.AzureSpatialAnchors/Scripts/AnchorModuleScript.cs

```diff
@@ -0,0 +1,487 @@
+                Debug.LogFormat("args.Anchor {0} Pose {1},{2},{3}", args.Anchor.Identifier, anchorPose.position.x,anchorPose.position.y,anchorPose.position.z);
+                
+                anchorPose = currentCloudAnchor.GetPose();
+                Debug.LogFormat("currentCloudAnchor {0} Pose {1},{2},{3}", currentCloudAnchor.Identifier, anchorPose.position.x,anchorPose.position.y,anchorPose.position.z);
+                
+                
+            }
+            currentCloudAnchor = args.Anchor;
+
+            QueueOnUpdate(() =>
+            {
+                Debug.Log($"Azure anchor located successfully");
+
+                // Notify AnchorFeedbackScript
+                OnASAAnchorLocated?.Invoke();
+
+#if WINDOWS_UWP || UNITY_WSA
+                // HoloLens: The position will be set based on the unityARUserAnchor that was located.
+
+                // Notify AnchorFeedbackScript
+                OnCreateLocalAnchor?.Invoke();
+
+                // On HoloLens, if we do not have a cloudAnchor already, we will have already positioned the
+                // object based on the passed in worldPos/worldRot and attached a new world anchor,
+                // so we are ready to commit the anchor to the cloud if requested.
+                // If we do have a cloudAnchor, we will use it's pointer to setup the world anchor,
+                // which will position the object automatically.
+                if (currentCloudAnchor != null)
+                {
+                    gameObject.AddComponent<CloudNativeAnchor>().CloudToNative(currentCloudAnchor);
+                    
+                    OnCreateLocalAnchor?.Invoke();
+                }
+
+#elif UNITY_ANDROID || UNITY_IOS
+                Pose anchorPose = Pose.identity;
+                anchorPose = currentCloudAnchor.GetPose();
+
+                Debug.Log($"Setting object to anchor pose with position '{anchorPose.position}' and rotation '{anchorPose.rotation}'");
+                transform.position = anchorPose.position;
+                transform.rotation = anchorPose.rotation;
+
+                // Create a native anchor at the location of the object in question
+                gameObject.CreateNativeAnchor();
+
+                // Notify AnchorFeedbackScript
+                OnCreateLocalAnchor?.Invoke();
+
+#endif
+            });
+        }
+        else
+        {
+            QueueOnUpdate(new Action(() => Debug.Log($"Attempt to locate Anchor with ID '{args.Identifier}' failed, locate anchor status was not 'Located' but '{args.Status}'")));
+        }
+    }
+    #endregion
+
+    #region Internal Methods and Coroutines
+    private void QueueOnUpdate(Action updateAction)
+    {
+        lock (dispatchQueue)
+        {
+            dispatchQueue.Enqueue(updateAction);
+        }
+    }
+
+    IEnumerator GetSharedAzureAnchorIDCoroutine(string sharingPin)
+    {
+        string url = "http://167.99.111.15:8090/file-uploads/static/file." + sharingPin.ToLower();
```

Comment:
Same commend as for `ShareAzureAnchorIdToNetwork`

What's this hardcoded IP address about? Seems like this should be a serialized property.

Actually...I don't think `GetSharedAzureAnchorIDCoroutine` ever called. Can you verify (search MRTK3 scripts, prefabs, and scenes). If not used remove this function please

Reply:
Removed